### ISSUE TITLE
[3.x]: Multiple icons missing in CP

### ISSUE THREAD

**Issue by proimage** [craftcms/cms#11619]
### Description
Recently (I wish I could point to a specific update but I cannot), I've noticed that multiple icons are missing from the Craft CP (EDIT: in Chrome Version 103.0.5060.114 (Official Build) (64-bit)):

![image](https://user-images.githubusercontent.com/1027523/179515502-8a391a94-660c-4c04-8761-11c5011971ea.png)

At first I thought it was all first-party icons in the main menu, but then I saw that GraphQL was still intact. Here's the rendered source from devtools of the GraphQL and Utilities menu items:

![image](https://user-images.githubusercontent.com/1027523/179516176-c2298008-678c-4d00-ab92-d1dd8a0059f0.png)

I have cleared the Control panel resources cache, to no effect. There are no errors in the console.

EDIT: Additional findings... this is not happening in Firefox. It _does_ happen in a Chrome Incognito window.

### Steps to reproduce

1. No idea... it just _is_.

### Expected behavior

I'd expect all the usual icons to show up.

### Actual behavior

The SVG for many icons isn't being included/rendered/injected into the source code.

### Craft CMS version

3.7.43

### PHP version

7.4.3

### Operating system and version

Windows 11 Version 21H2 (OS Build 22000.795)

### Database type and version

MySQL 5.6.47.0

### Image driver and version

Imagick 3.4.4 (ImageMagick 6.9.10-23)

### Installed plugins and versions

- Azure Blob Storage	1.0.1
- Button Box	2.0.4
- Colorit	1.1.2.1
- Cookie Consent Banner	1.2.9
- Cookies	1.1.15
- Craft Color Mixer	1.0.9
- Field Manager	2.2.4
- Freeform	3.13.10
- Icon Picker	1.1.12
- Imager X	v3.5.8
- Inventory	2.1.1
- Isolate	1.4.4
- Linkit	1.1.12.1
- LJ Range Slider	1.0.0
- Maps	3.9.3
- MatrixMate	1.4.3
- Minify	1.2.11
- Mix	1.5.2
- Navigation	1.4.26
- Neo	2.13.10
- oEmbed	1.3.15
- Postmark	2.1.0
- Redactor	2.10.8
- SEO	3.7.4
- Super Dynamic Fields	1.0.7
- Super Table	2.7.2
- Table Maker	2.0.1
- Tags	1.0.9
- Wordsmith	3.3.1.1

**Comment by brandonkelly** [comment]
A couple people have reported this in Discord as well, but we haven’t been able to get to the bottom of it. It appears to be something to do with a recent update to Chrome, however we are unable to reproduce it ourselves using the same version.

**Comment by proimage** [comment]
Hmm. I'd love to do what I can to troubleshoot this; I _am_ encountering this problem across all the Craft 3 sites I manage. FWIW, it doesn't seem to happen on a Craft 2.9.2 site I have access to.

I've disabled the three Chrome extensions I had that were allowed in Incognito mode, but it didn't make any difference.

It seems telling to me that the `svg` code itself is missing from the compiled source view in devtools. I take it the SVGs are getting inserted via JS for some reason, rather than by Twig?

**Comment by stephanwergles** [comment]
For me, this error has also been occurring recently. My feeling is that it happened shortly after an update of Chrome and that across different installations. I can also confirm that the error occurs in the incognito window, Firefox and all other browsers have no problems. In my case, the console does not show any traceable errors either.

**Comment by watarutmnh** [comment]
The same issue here with Chrome Version 103.0.5060.134, but no issue with Chrome Canary Version 105.0.5190.0.
Both are the latest version.

**Comment by thupsi** [comment]
Can confirm this is an issue also in Craft 4, in Chrome 103.0.5060.134, Windows 11

**Comment by thupsi** [comment]
Yet another update: I can't reproduce the issue on my laptop (up to now I was testing on my desktop), running the exact same version of Windows, Chrome, Craft etc. Hmm, very weird.

**Comment by proimage** [comment]
I think the first thing we need to figure out is how those `<svg>` icons end up on the page: are they injected by Javascript, or does Twig build the page with them as part of the source code?

**Comment by twitcher07** [comment]
I can confirm this is happening to me in Chrome Version 103.0.5060.114. This Craft sidebar looks like this for me:
![image](https://user-images.githubusercontent.com/8517321/180062107-6c83870f-e35c-41dd-bb15-1d360c487d0d.png)

In Firefox version 102.0.1 the icons seem to be working just fine:
![image](https://user-images.githubusercontent.com/8517321/180062220-8f82cebe-5cf8-4dee-bdc0-a1b2648695d4.png)

My Craft version is 3.7.30.1.

**Comment by thupsi** [comment]
> I think the first thing we need to figure out is how those `<svg>` icons end up on the page: are they injected by Javascript, or does Twig build the page with them as part of the source code?

The template file is this: https://github.com/craftcms/cms/blob/75d999d9a5d5351186d02a6a8286f89ae4c035b3/src/templates/_layouts/components/global-sidebar.twig

At about line 31 you can see that the icon is either an `<svg>`, or a font icon. In our case the SVGs (for example thr GraphQL icon) are displaying correctly, but the font icons are not.

**Comment by thupsi** [comment]
OK, so, investigating further, try this:
Open cp.css via devtools and add this line at the end of the file:
```
[data-icon="section"]{ }
```

As you can see, it's just an empty rule, referencing the `data-icon` attribute of the `entries` nav item. 

Result:
<img width="118" alt="image" src="https://user-images.githubusercontent.com/19857599/180155645-77193145-1b56-48c1-9844-4c099804cbc7.png">

That's crazy! It's like the browser doesn't match elements with the `[data-icon]` attribute selector unless you include a even an empty rule with a value match, like `[data-icon=value]`. I don't know if it's JS related, but it keeps getting weirder 🙃

**Comment by proimage** [comment]
Ok, you wanna see something even weirder? It seems like whatever the value is of that `[data-icon="..."]` selector is, is treated as a wildcard. So it fixes the icons for anything containing the letter(s) specified. I tried with the most common letters, vowels:

`[data-icon="a"]`:
![image](https://user-images.githubusercontent.com/1027523/180222112-be53d473-15d3-49da-a8e7-46cf3dd32e75.png)

`[data-icon="e"]`:
![image](https://user-images.githubusercontent.com/1027523/180222173-cef0f86a-9cb3-4851-a4dc-7e1f099e6af6.png)

`[data-icon="i"]`:
![image](https://user-images.githubusercontent.com/1027523/180222246-1d5d69bd-5613-469a-a9e7-25f63ec54574.png)

...etc.

This bug is almost as perplexing as that time ~15+ years ago when I was building my brother's PC for him, and powered the thing on, only to hear absolutely nothing... no case fans, no CPU fan, no HDD spinning, and no PSU fan... nothing was physically moving. And yet somehow, it POSTed! 🤪

**Comment by turnstylerj** [comment]
Super weird, I just noticed this after updating to Chrome 103.0.5060.134. Happening on multiple sites I'm sure were showing all the icons prior to today with the same version of Craft/plugins.

**Comment by proimage** [comment]
Font icons are no longer best practice these days (were they ever?), so perhaps at the very least this should be seen as a reason to switch to SVGs across the board in Craft? @brandonkelly

**Comment by alexr8** [comment]
One additional annoying part of this bug is the blue globe icons don't show up in the entries table view, so you can't open an entry on the front end without going into the entry. (these icons):
<img width="68" alt="image" src="https://user-images.githubusercontent.com/35040195/180400936-7212751d-93b1-4c1d-b391-fd14ebbfb039.png">
Can also confirm its happening on craft 3 and 4 sites for me, only in chrome, currently v103.0.5060.134 but it happened on the previous version i had too before updating.

**Comment by bymayo** [comment]
Yeah we're also experiencing this issue, it looks specific to `Chrome Version 103.0.5060.134 (Official Build) (arm64)`

Plugin icons load fine, it's just Craft's own icons (Entries, Utilities etc) that don't load.

**Comment by croxton** [comment]
Also seeing this bug and it affects all versions of Craft 3  that I've tried, unfortunately.

`Chrome Version 103.0.5060.134 (Official Build) (arm64)`

I've also seen some hints of odd behaviour with CSS generally when hard refreshing various webpages in this latest version of Chrome - very occasionally, CSS styles don't get applied or only some styles. Which is troubling!

**Comment by croxton** [comment]
Quick workaround:

Install the Control Panel CSS plugin for Craft CMS:
https://github.com/doublesecretagency/craft-cpcss

**Craft 4:**
`composer require doublesecretagency/craft-cpcss`

**Craft 3:**
`composer require doublesecretagency/craft-cpcss:2.4`

Install the plugin, go to it's settings and paste in this CSS to restore all your icons:
```
[data-icon]::before {
	content: attr(data-icon);
}

.icon::before, .menu ul.padded li a.sel::before, .menu .flex.padded.sel::before, .texticon::before, .element::before, #help::before, .secure::before, .insecure::before, .go::after, .required::after, #preview-btn::before, #share-btn::before, [data-icon]::before, [data-icon-after]::after {
	font-family: "Craft";
	speak: none;
	-webkit-font-feature-settings: "liga","dlig";
	-ms-font-feature-settings: "liga","dlig";
	-o-font-feature-settings: "liga","dlig";
	font-feature-settings: "liga","dlig";
	text-rendering: optimizeLegibility;
	font-weight: normal;
	font-variant: normal;
	text-transform: none;
	line-height: 1;
	direction: ltr;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	display: inline-block;
	text-align: center;
	font-style: normal;
	vertical-align: middle;
	word-wrap: normal !important;
	-webkit-user-select: none;
	user-select: none;
	opacity: .8;
}
```

**Comment by croxton** [comment]
Note that the bug appears to be that Chrome does not apply any rules that contain [data-icon] in cp.css.

**Comment by digason** [comment]
Icons are not showing in the latest Chromium dev build:
`Version 106.0.5196.0 (Developer Build) (x86_64)`

I suspect this issue will show up on other Chromium based browsers at some point, like Edge and Opera.

**Comment by mmikkel** [comment]
FTR, I'm *not* able to reproduce on Chrome `103.0.5060.134 (Official Build) (arm64)` on macOS 12.4.

**Comment by proimage** [comment]
> FWIW, I'm _not_ able to reproduce (i.e. icons are visible) on Chrome `103.0.5060.134 (Official Build) (arm64)` on macOS 12.4.

Could you try under Windows (using browserstack or parallels or whatever)? Perhaps it's a Chrome + Windows issue.

**Comment by mmikkel** [comment]
> > FWIW, I'm _not_ able to reproduce (i.e. icons are visible) on Chrome `103.0.5060.134 (Official Build) (arm64)` on macOS 12.4.
> 
> Could you try under Windows (using browserstack or parallels or whatever)? Perhaps it's a Chrome + Windows issue.

The issue has already been [seen on Mac](https://github.com/craftcms/cms/issues/11619#issuecomment-1192335961), so that doesn’t seem to be the case. The problem seems more complicated than just a platform/version thing; maybe it’s something to do w/ Chrome configuration settings (perhaps some specific combination of security options or the like?)

**Comment by adrianjean** [comment]
I too am seeing this — on nearly every CraftCMS site I update to the latest version. I also notices other icons are missing on other system pages like down arrows and such. 

**Platform:** MacOS 12.4 (21F79)
**Browser:** Chrome - 103.0.5060.134 (Official Build) (arm64)
**Craft:** Craft CMS 3.7.48 Pro

**Plugins Installed (All up to date):**
AsyncQueue
Button Box
Control Panel CSS (I tested without this as well)
Control Panel Nav (I tested without this as well)
Dashboard Begone
Eager Beaver
Enupal Backup
Expanded Singles
Formie
Freeform
Knock Knock
Redactor
Redactor Anchors
Redactor Custom Styles
Retour
SEOmatic
SendGrid
Sprig
Super Table

**Comment by turnstylerj** [comment]
![Screen Shot 2022-07-24 at 2 26 47 PM](https://user-images.githubusercontent.com/42223501/180666588-f9d2fc2d-92b2-4bf3-8867-70b9bcc8707e.png)

I just realized I'm also missing the field handle icons and translatable field icons in settings, as well as the translatable field icons in entry views for authors, which feels like a much bigger deal.

Also on macOS v12.2.1.

**Comment by croxton** [comment]
Not sure how this helps but I've discovered that you can have a maximum of ~44 individual CSS rules in cp.css that reference `[data-icon]`. Any more and the icons disappear. It is the number of rules rather than anything specific about them that matters.

It would suggest that Chrome has a maximum quota for the number of matches for a given selector in a single stylesheet.

**Comment by croxton** [comment]
Resetting Chrome restored the icons for me, on `Version 103.0.5060.134 (Official Build) (arm64)`

Go to `chrome://settings/` and click `Reset settings` from the menu.

**Comment by alexr8** [comment]
As far as i know I've not changed anything (in terms of updating chrome, resetting settings, changing craft versions or updating my OS in any way), and my icons seem to appearing now just fine (questioning my sanity at this point 😅). Anyone else?

**Comment by iamtompickering** [comment]
@alexr8 same thing has happened to me today! 

I've not touched anything on my machine and suddenly everything is back to normal.

**Comment by croxton** [comment]
Clearly what happened was my Chrome install was in a quantum superposition with every other Chrome instance worldwide, so when I reset it, it collapsed the wave function reducing them all to a single eigenstate. That seems to be the only logical explanation 🙃

**Comment by proimage** [comment]
What the actual heck. The problem has disappeared for me as well. The only Chrome difference is that the version went from 103.0.5060.**114** to 103.0.5060.**134**.

Here's the release post: https://chromereleases.googleblog.com/2022/07/stable-channel-update-for-desktop_19.html
Here is the diff: https://chromium.googlesource.com/chromium/src/+log/103.0.5060.114..103.0.5060.134?pretty=fuller&n=10000

Anyone see anything incriminating?

**Comment by croxton** [comment]
> What the actual heck. The problem has disappeared for me as well. The only Chrome difference is that the version went from 103.0.5060.**114** to 103.0.5060.**134**.
> 
> Here's the release post: https://chromereleases.googleblog.com/2022/07/stable-channel-update-for-desktop_19.html Here is the diff: https://chromium.googlesource.com/chromium/src/+log/103.0.5060.114..103.0.5060.134?pretty=fuller&n=10000
> 
> Anyone see anything incriminating?

Mine was already at `103.0.5060.134` when I first noticed the problem 🤷🏻‍♂️

**Comment by alexr8** [comment]
> What the actual heck. The problem has disappeared for me as well. The only Chrome difference is that the version went from 103.0.5060.**114** to 103.0.5060.**134**.

My version is exactly the same as when i had the issue (103.0.5060.134), so not sure its the new release thats changed anything 🤔

**Comment by adrianjean** [comment]
Ok... so... SAME. 
I did nothing except go to bed, then wake up, and it's all working now.  ¯\_(ツ)_/¯

**Comment by bymayo** [comment]
@adrianjean It's amazing when bugs fix themselves, whilst you sleep. It just doesn't happen near enough 😂

**Comment by turnstylerj** [comment]
Very strange, this is still an issue for me on Chrome 103.0.5060.134 on macOS 12.2.1.

**Comment by turnstylerj** [comment]
Update: Completely quitting and relaunching Chrome has resolved this for me. Same exact versions of everything. 🤷

**Comment by proimage** [comment]
> @adrianjean It's amazing when bugs fix themselves, whilst you sleep. It just doesn't happen near enough 😂

That's the absolute worst case scenario IMO—when you don't know why a bug is happening in the first place gets mysteriously resolved without anything that you're aware of changing? Ugh.

**Comment by stephanwergles** [comment]
How crazy is that? I still have Chrome version 103.0.5060.134 and overnight the problem is solved. macOS 12.4

**Comment by twitcher07** [comment]
> Update: Completely quitting and relaunching Chrome has resolved this for me. Same exact versions of everything. 🤷

I can confirm on my MacOS chrome version 103.0.5060.134 that quitting chrome and relaunching has brought the icons back for me.

![image](https://user-images.githubusercontent.com/8517321/181033459-533733d0-2825-48de-8ae8-a2ff99f0ce2a.png)

**Comment by benediktblank** [comment]
> Update: Completely quitting and relaunching Chrome has resolved this for me. Same exact versions of everything. 🤷

It suddenly started working again 2 days ago, yeah. Quitting and relaunching did nothing for me before that day.
Something else must have changed. But Chrome version is indeed the same one.

**Comment by robzor** [comment]
OS X Catalina - 10.15.7
Chrome - 103.0.5060.134

Had this today, doing a proper Quit and relaunch of Chrome fixed it for me.

**Comment by DavidKabelitz** [comment]
Hey guys, i have the same Chrome issue here. I restarted Chrome tons of time but still icons are missing. What do mean with "proper Quit"? Also some Ajax things are not working. Looks like a random bug on chrome
<img width="156" alt="Bildschirmfoto 2022-08-09 um 10 22 05" src="https://user-images.githubusercontent.com/28010175/183601288-c6b6aead-24ce-4d94-ab36-cf014df24abd.png">
.

**Comment by darylknight** [comment]
> Hey guys, i have the same Chrome issue here. I restarted Chrome tons of time but still icons are missing. What do mean with "proper Quit"? Also some Ajax things are not working. Looks like a random bug on chrome <img alt="Bildschirmfoto 2022-08-09 um 10 22 05" width="156" src="https://user-images.githubusercontent.com/28010175/183601288-c6b6aead-24ce-4d94-ab36-cf014df24abd.png"> .
@DavidKabelitz 

I think your specific issue is unrelated. The rest of us had the icons completely missing. Typically if you're seeing the alt text of the icon, I've found that it's because your site isn't redirecting to a single version of the site correctly (i.e, you can access the site from both https:// or http://, and/or the non-www version or the www version. You need to make sure that you redirect traffic to the exact version you have listed in .env.

**Comment by DavidKabelitz** [comment]
@darylknight yes you are right, our local server runs on localhost and an ip, with the ip its working but not with the hostname and it was related to your comment. But it just happens at Chrome, that's strange.

**Comment by darylknight** [comment]
@DavidKabelitz Yep, separate issue. You need to access your site on only the URL that matches your .env file. Feel free to ping me or the Discord server about this so we don't clutter this github issue.

**Comment by DavidKabelitz** [comment]
Yes i know, we just recognized it after the comment here
